cpp/modules/deck.gl/core/src/lib/deck.cc

```diff
@@ -72,6 +72,10 @@ Deck::Deck(std::shared_ptr<Deck::Props> props)
       lumagl::utils::createBuffer(this->animationLoop->device(), sizeof(ViewportUniforms), wgpu::BufferUsage::Uniform);
 }
 
+auto Deck::make(probegl::Error* error, std::shared_ptr<Deck::Props> props) -> std::optional<Deck> {
+  return probegl::executeThrowable<Deck>([&]() { return Deck{props}; }, error);
```

Comment:
Yes this looks exactly right. As for naming, `executeThrowable`  -> `exceptionToError` or just `catchError()` since a short name is desirable for such a commonly used helper?